### ISSUE TITLE
test speed 

### ISSUE THREAD

**Issue by June-hui** [Little-Podi/GRM#10]
hello ! i want to know how you  test speed on your model (45fps on got10k)?   can u provide the "profile_model.py" like other tracker? thank u!

**Comment by Little-Podi** [comment]
Hi, I directly took the average FPS printed during inferecne as the running speed of my tracker. The `profile_model.py` needs the inputs manually prepared, and I do have a messy version. It would be an useful reference, whereas normally it will provide a faster result. Besides, the amount of FLOPs also cannot truly reveal the computation latency of algorithms. The most simple and straightforward way is still running them on the same machine. If you want it, I think you can smoothly modify a script from [STARK](https://github.com/researchmm/Stark/blob/main/tracking/profile_model.py) or [OSTrack](https://github.com/botaoye/OSTrack/blob/main/tracking/profile_model.py).

**Comment by YmShan** [comment]
I wrote a `profile_model.py` script that is not very well-structured. The code is as follows.

```python
import os
import sys
import argparse
import torch
from lib.utils.misc import NestedTensor
from thop import profile
from thop.utils import clever_format
import time
import importlib

# Set project path
prj_path = os.path.join(os.path.dirname(__file__), '..')
if prj_path not in sys.path:
    sys.path.append(prj_path)

def parse_args():
    """
    Parse arguments for training.
    """
    parser = argparse.ArgumentParser(description='Parse args for training')
    parser.add_argument('--script', type=str, default='grm', choices=['ostrack'],
                        help='Training script name')
    parser.add_argument('--config', type=str, default='vitb_256_got_ep100', help='YAML configuration file name')
    args = parser.parse_args()
    return args

def evaluate_vit(model, template, search):
    """Speed Test and Model Profiling"""
    
    macs1, params1 = profile(model, inputs=(template, search),
                             custom_ops=None, verbose=False)
    macs, params = clever_format([macs1, params1], "%.3f")
    print('Overall MACs: ', macs)
    print('Overall Params: ', params)

    T_w = 500  # Warm-up iterations
    T_t = 1000  # Timed iterations
    print("Testing speed ...")
    
    torch.cuda.synchronize()
    with torch.no_grad():
        # Warm-up
        for _ in range(T_w):
            _ = model(template, search)
        start = time.time()
        for _ in range(T_t):
            _ = model(template, search)
        torch.cuda.synchronize()
        end = time.time()
    
    avg_latency = (end - start) / T_t
    print("Average Overall Latency: %.2f ms" % (avg_latency * 1000))
    print("FPS: %.2f fps" % (1. / avg_latency))

# Function to generate dummy data
def get_data(bs, sz):
    """Generate Random Input Data"""
    img_patch = torch.randn(bs, 3, sz, sz)
    att_mask = torch.rand(bs, sz, sz) > 0.5
    return NestedTensor(img_patch, att_mask)

if __name__ == "__main__":
    device = "cuda:0"
    torch.cuda.set_device(device)

    # Parse arguments
    args = parse_args()

    # Update configuration
    yaml_fname = 'experiments/grm/vitb_256_got_ep100.yaml'
    config_module = importlib.import_module('lib.config.grm.config')
    cfg = config_module.cfg
    config_module.update_config_from_file(yaml_fname)

    # Set values
    bs = 1
    z_sz = cfg.TEST.TEMPLATE_SIZE
    x_sz = cfg.TEST.SEARCH_SIZE

    # Import and build model
    model_module = importlib.import_module('lib.models')
    model_constructor = model_module.build_grm
    model = model_constructor(cfg, training=False)

    # Generate input data
    template = torch.randn(bs, 3, z_sz, z_sz)
    search = torch.randn(bs, 3, x_sz, x_sz)

    # Move to device
    model = model.to(device)
    template = template.to(device)
    search = search.to(device)

    # Evaluate Model
    merge_layer = cfg.MODEL.BACKBONE.MERGE_LAYER
    if merge_layer <= 0:
        evaluate_vit(model, template, search)
    else:
        raise NotImplementedError
